opentelemetry-sdk/src/logs/log_emitter.rs

```diff
@@ -45,48 +45,15 @@ pub struct LoggerProvider {
     inner: Arc<LoggerProviderInner>,
 }
 
-/// Default logger name if empty string is provided.
-const DEFAULT_COMPONENT_NAME: &str = "rust.opentelemetry.io/sdk/logger";
-
 impl opentelemetry::logs::LoggerProvider for LoggerProvider {
     type Logger = Logger;
 
-    /// Create a new versioned `Logger` instance.
-    fn versioned_logger(
-        &self,
-        name: impl Into<Cow<'static, str>>,
-        version: Option<Cow<'static, str>>,
-        schema_url: Option<Cow<'static, str>>,
-        attributes: Option<Vec<opentelemetry::KeyValue>>,
-    ) -> Logger {
-        let name = name.into();
-        let component_name = if name.is_empty() {
-            Cow::Borrowed(DEFAULT_COMPONENT_NAME)
```

Comment:
Are we changing the behavior to not override empty names with the default component name?

Reply:
I suggest we move the `signal` methods to `named_signal` and let the trait implementers handle the logic around empty names in their implementations of `signal`:

```rust
trait LoggerProvider {
    fn named_logger(&self, name: impl Into<Cow<'static, str>>) -> Self::Logger {
        let scope = InstrumentationScope::builder(name).build();
        self.logger_with_scope(scope)
    }
    
    fn logger(&self, name: impl Into<Cow<'static, str>>) -> Self::Logger;

    fn logger_with_scope(&self, scope: InstrumentationScope) -> Self::Logger;
}
```

Or maybe we should simply get rid of `named_signal`?

Reply:
I went with the overrides in a separate commit, let me know what you think!